### PR TITLE
webapp/tip: switch to Antd tooltip

### PR DESCRIPTION
**Pull request by haraldschilly**
# Description

This is about replacing "Tip" with antd components. However, the hub doesn't run. The share server part imports r_misc, and that fails. One issue was about importing css in the antd.ts file, but I did split this up and the error below is the next one I encountered. I tried various things, including tweaking tsconfig etc. It's really weird, because in other r_misc files like "icon", there is also `import * as React from 'react';` working fine (?). 

However, catching that error seems to be possible, i.e. share server starts and I log the error message like this `Tooltip&Popover cannot be imported -- SyntaxError: Unexpected token *` 

I think this is ready for review, otherwise it's getting too long. I also tried the "topLeft" slightly offset placements for the chat button or kernel status. Looks better than before. The scaling is to make it compatible with the size properties we have, but it jumps a bit around. Feels like a bug in Antd to me.

```
import * as React from 'react';
       ^

SyntaxError: Unexpected token *
    at Module._compile (internal/modules/cjs/loader.js:311:1)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:311:1)
    at Module.load (/home/user/cocalc/src/smc-hub/node_modules/coffeescript/lib/coffeescript/register.js:75:9)
    at tryModuleLoad (internal/modules/cjs/loader.js:311:1)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:311:1)
    at require (internal/modules/cjs/helpers.js:43:1)
    at Object.<anonymous> (/home/user/cocalc/src/cocalc-ui/antd.ts:22:1)
[...]
```

---

travis CI now complains about

```
ERROR in /home/travis/build/sagemathinc/cocalc/src/smc-webapp/r_profile_image.tsx
./smc-webapp/r_profile_image.tsx
[tsl] ERROR in /home/travis/build/sagemathinc/cocalc/src/smc-webapp/r_profile_image.tsx(36,15)
      TS2503: Cannot find namespace 'ReactCrop'.
ERROR in /home/travis/build/sagemathinc/cocalc/src/smc-webapp/r_profile_image.tsx
./smc-webapp/r_profile_image.tsx
[tsl] ERROR in /home/travis/build/sagemathinc/cocalc/src/smc-webapp/r_profile_image.tsx(446,14)
      TS2503: Cannot find namespace 'ReactCrop'.
```

but I have no idea how this can happen. my only idea is the small TS update.

however, updating react-image-crop to something higher than 6 has a breaking API change. i bet a version 8 will work fine, but needs more work...

well, I'm trying to revert npm related updated, but so far no luck. 

# Testing Steps
hover over different buttons and areas. e.g. project title, file, kernel status, assignment button in courses, side-chat button, etc.

# Screenshots

some teaser

Screenshot from 2019-10-15 18-58-17
Screenshot from 2019-10-15 18-57-27
Screenshot from 2019-10-15 18-34-58

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):
- [ ] No debugging console.log messages.
- [ ] All new code is actually used.
- [ ] Non-obvious code has some sort of comments.

Front end:
- [ ] Restart at least one project and check its `~/.smc/local_hub/local_hub.log`
- [ ] Completely restart Webpack with `./w` in `/src`
- [ ] Completely restart the hub by killing and restarting `./start_hub.py` in `/src/dev/project`
- [ ] Screenshots if relevant.

### CODE REVIEW

File: src/smc-webapp/customize.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ Site Customize -- dynamically customize the look of SMC for the client.
 
 import { redux, Redux, rclass, rtypes } from "./app-framework";
 import * as React from "react";
-import { Loading } from "./r_misc";
+import { Loading } from "./r_misc/loading";
```
[the remark this answers is not in the document]
Comment (haraldschilly) [reply src/smc-webapp/customize.tsx]: I don't understand what's being easier or not. IIRC the overall problem was that loading the whole r_misc module wasn't possible and picking only what I needed did work. I'll undo this, maybe there is a better solution.

File: src/cocalc-ui/index.ts [hunk 2]
```diff
@@ -1,45 +1 @@
-import "antd/es/tooltip/style/css";
```
[the remark this answers is not in the document]
Comment (haraldschilly) [reply src/cocalc-ui/index.ts]: Well, why are we splitting things up in smaller files? There is a file just for antd now. The css definitions are part of webpack's "css" chuck, which also holds all the other CSS definitions for the cocalc UI. So, that's why it makes sense to me to keep all CSS related aspects in one place.

File: src/smc-webapp/r_misc/tip.tsx [hunk 3]
```diff
@@ -1,50 +1,69 @@
...
     super(props);
-    this.state = { display_trigger: false };
+    this.state = {};
   }
 
   static defaultProps = {
     placement: "right",
     delayShow: 500,
-    delayHide: 0,
+    delayHide: 100, // was 0, but .1 is the Antd default
```
[the remark this answers is not in the document]
Comment (haraldschilly) [reply src/smc-webapp/r_misc/tip.tsx]: yes, that's confusing, this should be all milliseconds

File: src/tsconfig.json [hunk 4]
```diff
@@ -16,6 +16,7 @@
     "smc-project",
     "smc-hub",
     "smc-util-node",
+    "cocalc-ui",
```
[the remark this answers is not in the document]
Comment (haraldschilly) [reply src/tsconfig.json]: Well, it makes sense to me to treat it like the other modules, that's all.

File: src/webapp-css.coffee [hunk 5]
```diff
@@ -32,6 +31,9 @@ require('octicons/octicons/octicons.css')
 # custom cocalc icon font
 require('webapp-lib/cocalc-icons-font/style.css')
 
+# cocalc-ui
+require('cocalc-ui/index-css.ts')
```
[the remark this answers is not in the document]
Comment (haraldschilly) [reply src/webapp-css.coffee]: This file is the root for webpack's css chunck definition. All these cocalc-ui related styles are loaded here, that's all.

**Comment by haraldschilly** [comment]
> then the whole tip shrinks like 10% afterwards, which feels very broken

the interface defines 4 sizes which Antd doesn't have. that's why the code adds some style, but I don't know why it doesn't add it before it is shown. I should check if there is an issue about that.

More general, this branch is blocked in any case, because of that `ReactCrop` problem. I don't want to spend more with it as long as we use the old version.

File: src/smc-webapp/share/public-paths-browser.tsx [hunk 6]
```diff
@@ -5,7 +5,7 @@ Share server top-level landing page.
 import { List, Map } from "immutable";
 import { Component, React, Rendered } from "../app-framework";
 import { encode_path } from "smc-util/misc";
-const { TimeAgoElement } = require("../r_misc");
+const { TimeAgoElement } = require("../r_misc/old-coffee-code-index");
```
[the remark this answers is not in the document]
Comment (williamstein) [reply src/smc-webapp/share/public-paths-browser.tsx]: >  I'll undo this.

But if it doesn't work, how can you undo it.    To clarify, when I wrote "Why???" I literally meant "why?".  If this code `const { TimeAgoElement } = require("../r_misc");` that I wrote doesn't work now due to a mysterious bug, and you can't resolve the bug, I'm OK with writing `const { TimeAgoElement } = require("../r_misc/old-coffee-code-index");` as long as there is also a comment explaining what is going on, preferably with a link to a github issue.    Just document it.  Somebody is definitely going to come across this later...

**Comment by williamstein** [comment]
No longer blocked due to reactcrop, which is merged.

**Pull request merged**